src/orion/storage/legacy.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.storage.legacy` -- Legacy storage
+=============================================================================
+
+.. module:: legacy
+   :platform: Unix
+   :synopsis: Old Storage implementation
+
+"""
+
+from typing import Optional
+
+from orion.core.io.convert import JSONConverter
+from orion.core.io.database import Database
+from orion.core.worker.trial import Trial
+from orion.storage.base import BaseStorageProtocol
+
+
+class Legacy(BaseStorageProtocol):
+    """Legacy protocol, forward most request to experiment"""
+
+    def __init__(self, uri=None):
+        """INIT METHOD"""
+        self.converter = JSONConverter()
+        self._db = Database()
+        self._setup_db()
+
+    def _setup_db(self):
+        """Database index setup"""
+        self._db.ensure_index('experiments',
+                              [('name', Database.ASCENDING),
+                               ('metadata.user', Database.ASCENDING)],
+                              unique=True)
+        self._db.ensure_index('experiments', 'metadata.datetime')
+
+        self._db.ensure_index('trials', 'experiment')
+        self._db.ensure_index('trials', 'status')
+        self._db.ensure_index('trials', 'results')
+        self._db.ensure_index('trials', 'start_time')
+        self._db.ensure_index('trials', [('end_time', Database.DESCENDING)])
+
+    def create_experiment(self, config):
+        """See :func:`~orion.storage.BaseStorageProtocol.create_experiment`"""
+        return self._db.write('experiments', config)
+
+    def update_experiment(self, experiment: 'Experiment', fields: Optional[dict] = None,
```

Comment:
Why should we support both `fields` and `kwargs`? `fields` can easily be avoided with `foo(**data)`.